http3/server.go

```diff
@@ -217,7 +218,8 @@ type Server struct {
 	mutex     sync.RWMutex
 	listeners map[*QUICEarlyListener]listenerInfo
 
-	closed bool
+	closed      bool
+	connections map[*quic.Connection]func()
```

Comment:
Better: Use the tracing ID as a map key.

Reply:
But the methods on the `quic.Connection` needs to be called. To properly stop server from accepting new streams at the grace period's end. Malicious client may attempt to open new streams even if goaway is received. [Goaway itself doesn't close a connection](https://www.rfc-editor.org/rfc/rfc9114.html#name-goaway).

Reply:
> Better: Use the tracing ID as a map key.

Update: the tracing ID has now been deprecated, so we shouldn't use that one.